### ISSUE TITLE
URI::createFromEnvironment mistake?

### ISSUE THREAD

**Issue by egonom** [slimphp/Slim#2195]
Hi Guys!
In my project the script directory name is "public".

> +app
> -public
>  index.php
> +vendor

When I defined a route which started with 'public', then the createFromEnvirontment method chunks this part:
the route: 
`$app->get('/public_idea/{idea_id}',`	'App\Action\...');

and the request in App.php:
 /var/.../vendor/slim/slim/Slim/App.php on line 367
```
Slim\Http\Request Object
(
    [method:protected] => 
    [originalMethod:protected] => GET
    [uri:protected] => Slim\Http\Uri Object
        (
            [scheme:protected] => http
            [user:protected] => 
            [password:protected] => 
            [host:protected] => xxx
            [port:protected] => 80
            [basePath:protected] => /public
            [path:protected] => _idea/1 <-------------------------------
            [query:protected] => 
            [fragment:protected] => 
        )
```
If I tailed the requestScriptDir with a slash, it improved:

```
//        } elseif ($requestScriptDir !== '/' && stripos($requestUri, $requestScriptDir) === 0) {
        } elseif ($requestScriptDir !== '/' && stripos($requestUri, $requestScriptDir.'/') === 0) {

```
If somebody using a failed url (included with the script directory name) the trailing slash will be there anyway, so imho this change will not cause trouble.
Best,
e.

**Comment by llvdl** [comment]
Can you provide the contents of `$env` that is passed to `URI::createFromEnvironment`?

**Comment by egonom** [comment]
Sorry, what kind of content would you like?
The request uri from $_SERVER is "/public_idea/1"
and the chunked path is "_idea/1" as above.

**Comment by mathmarques** [comment]
In theory, this is the problem:
```php
$environment = Environment::mock([
    'SCRIPT_NAME' => '/foo/index.php',
    'REQUEST_URI' => '/foo_bar',
    'HTTP_HOST' => 'example.com:8080',
    'SERVER_PORT' => 8080,
]);
```
But when the `/foo` won't be on `REQUEST_URI`?
For me the correct Request URI would be `'REQUEST_URI' => '/foo/foo_bar',`

**Comment by llvdl** [comment]
I created a `public/index.php` file with the following content

```php
<?php
var_dump($_SERVER);
```

and started a PHP built-in web server in the directory above `public`. I get the following output:

```
array (size=23)
  'SERVER_NAME' => string 'localhost' (length=9)
  'SERVER_PORT' => string '8000' (length=4)
  'REQUEST_URI' => string '/public/foo_bar' (length=15)
  'SCRIPT_NAME' => string '/public/index.php' (length=17)
  'PATH_INFO' => string '/foo_bar' (length=8)
  'PHP_SELF' => string '/public/index.php/foo_bar' (length=25)
  'HTTP_HOST' => string 'localhost:8000' (length=14)
  ...
```

Perhaps other servers (Apache, Nginx) return a different value for REQUEST_URI?

**Comment by egonom** [comment]
I'm so sorry! It was my mistake, but interested:
I've got from our SA a test host with config pointed for the web directory. When I looked it, the SA were on holiday, so I created a simple .htaccess file, rewrite all request (without condition) for the /public directory. And it's remain...

So my SCRIPT_NAME was /public/index.php against /index.php (wo the .htaccess rewrite).

AND... so the URI::createFromEnvironment() do that, what I wrote: it chunks the part of route which eq the rewrited target directory's name.

Thank you for the help!
